shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -551,7 +566,7 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
               } else TypeRef(td, Nil)(tpr, ctx)
             case L(e) =>
               if (name.isEmpty || !name.head.isLower) e()
-              else typeNamed(tyLoc, name.head.toUpper.toString + name.tail) match {
+              else typeNamed(tyLoc, name.capitalize) match {
                 case R(td) => td.kind match {
                   case Cls => clsNameToNomTag(td)(tp(tyLoc, "class tag"), ctx)
                   case Trt => ???
```

Comment:
This is from the last commit but it should just be `trtNameToNomTag` here. Might as well implement it.

Reply:
Whoops, good catch!